hadoop-common-project/hadoop-common/src/test/java/org/apache/hadoop/conf/TestConfiguration.java

```diff
@@ -1062,6 +1062,38 @@ public void testRelativeIncludes() throws Exception {
     new File(new File(relConfig).getParent()).delete();
   }
 
+  @Test
+  public void testRelativeIncludesWithLoadingViaUri() throws Exception {
+    tearDown();
+    File configFile = new File("./tmp/test-config.xml");
+    File configFile2 = new File("./tmp/test-config2.xml");
+
+    new File(configFile.getParent()).mkdirs();
+    out = new BufferedWriter(new FileWriter(configFile2));
+    startConfig();
+    appendProperty("a", "b");
+    endConfig();
+
+    out = new BufferedWriter(new FileWriter(configFile));
+    startConfig();
+    // Add the relative path instead of the absolute one.
+    startInclude(configFile2.getName());
+    endInclude();
+    appendProperty("c", "d");
+    endConfig();
+
+    // verify that the includes file contains all properties
+    Path fileResource = new Path(configFile.toURI());
+    conf.addResource(fileResource);
+    assertEquals(conf.get("a"), "b");
```

Comment:
params are the wrong way round. not your fault, but there's no reason to replicate the existing issue.

Reply:
@steveloughran Thanks for the review! You mean we need to write here like that, right? (considering `expected` and `actual` order)
```java
assertEquals("b", conf.get("a"));
```

Reply:
exactly. Because Junit reports arg 1 as the 'expected value' in the exceptions it raises...if they are the wrong way round the error is misleading and it takes longer to debug.

Reply:
Thanks! Fixed.